File: 91-aws/02-rest-api-mysql/src/main/java/com/in28minutes/rest/webservices/restfulwebservices/security/SpringSecurityConfiguration.java
package com.in28minutes.rest.webservices.restfulwebservices.security;


import static org.springframework.security.config.Customizer.withDefaults;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.web.SecurityFilterChain;

@Configuration
public class SpringSecurityConfiguration {
	
	@Bean
	public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
		
//		1) All requests should be authenticated
		http.authorizeHttpRequests(
				auth -> auth
					.requestMatchers("/").permitAll() //#CHANGE
					.anyRequest().authenticated()
				);
//		2) If a request is not authenticated, a web page is shown
		http.httpBasic(withDefaults());
		
//		3) CSRF -> POST, PUT
		http.csrf(csrf -> csrf.disable());

		
		return http.build();
	}

}

File: 12-rest-api-kotlin/src/main/kotlin/com/in28minutes/rest/webservices/restfulwebservices/security/SpringSecurityConfiguration.kt
package com.in28minutes.rest.webservices.restfulwebservices.security

import org.springframework.context.annotation.Configuration

@Configuration
class SpringSecurityConfiguration {

//	@Bean
//	fun filterChain(http: HttpSecurity): SecurityFilterChain {
//
//		// 1) All requests should be authenticated
//		http.authorizeRequests { auth ->
//			auth.anyRequest().authenticated()
//		}
//		// 2) If a request is not authenticated, use http basic
//		http.httpBasic(withDefaults())
//
//		// 3) CSRF -> POST, PUT
//		http.csrf { csrf ->
//			csrf.disable()
//		}
//		// OR
//		// http.csrf(AbstractHttpConfigurer::disable)
//
//		return http.build()
//	}

}